lib/proxy.js

```diff
@@ -533,13 +533,16 @@ ReverseProxy.prototype.resolve = function (host, url, req) {
     promiseArray.push(this.resolvers[i].call(this,host, url, req));
   }
 
-  return Promise.all(promiseArray).then(function (data) {
-    var route = data[0];
-    if (route && (route = ReverseProxy.buildRoute(route))) {
-      // ensure resolved route has path that prefixes URL
-      // no need to check for native routes.
-      if (!route.isResolved || route.path === '/' || startsWith(url, route.path)) {
-        return route;
+  return Promise.all(promiseArray).then(function (resolverResults) {
+    while (resolverResults.length) {
+      var route = resolverResults.shift();
```

Comment:
for performance it would be better to rely on a  standard for-loop than in .shift()